packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,864 @@
+///   ),
+///   title: "Example Window",
+/// );
+/// runWidget(RegularWindow(
+///   controller: controller,
+///   child: MaterialApp(home: Container())));
+/// ```
+///
+/// When provided to a [RegularWindow] widget, widgets inside of the [child]
+/// parameter will have access to the [RegularWindowController] via the
+/// [WindowControllerContext] widget.
+///
+/// {@template flutter.widgets.windowing.experimental}
+/// Do not use this API in production applications or packages published to
+/// pub.dev. Flutter will make breaking changes to this API, even in patch
+/// versions.
+///
+/// This API throws an [UnsupportedError] error unless Flutter’s windowing
+/// feature is enabled by [isWindowingEnabled].
+///
+/// See: https://github.com/flutter/flutter/issues/30701.
+/// {@endtemplate}
+@internal
+abstract class RegularWindowController extends WindowController {
+  /// Creates a [RegularWindowController] with the provided properties.
+  /// Upon construction, the window is created for the platform.
+  ///
+  /// [contentSize] sizing requests for the window. This may not be honored by the platform
+  /// [title] the title of the window
+  /// [state] the initial state of the window
+  /// [delegate] optional delegate for the controller controller.
+  ///
+  /// {@template flutter.widgets.windowing.experimental}
+  /// Do not use this API in production applications or packages published to
+  /// pub.dev. Flutter will make breaking changes to this API, even in patch
+  /// versions.
+  ///
+  /// This API throws an [UnsupportedError] error unless Flutter’s windowing
+  /// feature is enabled by [isWindowingEnabled].
+  ///
+  /// See: https://github.com/flutter/flutter/issues/30701.
+  /// {@endtemplate}
+  @internal
+  factory RegularWindowController({
+    required WindowSizing contentSize,
+    String? title,
+    RegularWindowControllerDelegate? delegate,
+  }) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_windowingDisabledErrorMessage);
+    }
+
+    WidgetsFlutterBinding.ensureInitialized();
+    final WindowingOwner owner = WidgetsBinding.instance.windowingOwner;
+    final RegularWindowController controller = owner.createRegularWindowController(
+      contentSize: contentSize,
+      delegate: delegate ?? RegularWindowControllerDelegate(),
+    );
+    if (title != null) {
+      controller.setTitle(title);
+    }
+    return controller;
+  }
+
+  /// Creates an empty [RegularWindowController].
+  ///
+  /// {@template flutter.widgets.windowing.experimental}
+  /// Do not use this API in production applications or packages published to
+  /// pub.dev. Flutter will make breaking changes to this API, even in patch
+  /// versions.
+  ///
+  /// This API throws an [UnsupportedError] error unless Flutter’s windowing
+  /// feature is enabled by [isWindowingEnabled].
+  ///
+  /// See: https://github.com/flutter/flutter/issues/30701.
+  /// {@endtemplate}
+  @internal
+  @protected
+  RegularWindowController.empty();
```

Comment:
Is this just used for testing purposes?

If yes, should we add a `TODO(foo): Replace `@internal` with `@visibleForTesting` when this API is non-experimental`?

Also, @justinmc do we have a naming convention for these kinds of constructors?

Reply:
Gotcha, closing!